File: Array/LC509-FibonacciNumber.py
class Solution:
    def fib(self, N: int) -> int:
        # This is the Fibonacci sequence
        f = [0 for _ in range(31)]
        # Initialization
        f[0] = 0
        f[1] = 1
        # Loop through the Fibonacci sequence
        for i in range(2, N + 1):
            f[i] = f[i - 1] + f[i - 2]
        return f[N]

File: Leet2022/DP/LC509_Fibonacci_Number.cpp
class Solution {
public:
    int fib(int n) {
        if (n == 1 || n == 0) {
            return n;
        }
        int prev = 0, curr = 1;
        for (int i = 2; i <= n; ++i) {
            int tmp = prev + curr;
            prev = curr;
            curr = tmp;
        }
        return curr;
    }
};
